tests/sync/session.cpp

```diff
@@ -196,6 +196,19 @@ TEST_CASE("SyncSession: management by SyncUser", "[sync]") {
         session = user->session_for_url(server.base_url() + path);
         CHECK(session);
     }
+
+    SECTION("a user cannot create multiple sessions for the same URL") {
```

Comment:
Would it be worthwhile to add a FIXME to remove this test once we remove this restriction?